File: python/appcode_demo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys,os
import base64
import time
import json

from urlparse import urlparse
from com.aliyun.api.gateway.sdk import client
from com.aliyun.api.gateway.sdk.http import request
from com.aliyun.api.gateway.sdk.common import constant
import traceback
import urllib2
import base64

def get_img_base64(img_file):
    with open(img_file, 'rb') as infile:
        s = infile.read()
        return base64.b64encode(s) 


def predict(url, appcode, img_base64, kv_configure):
        param = {}
        param['image'] = img_base64
        if kv_configure is not None:
            param['configure'] = json.dumps(kv_configure)
        body = json.dumps(param)

        headers = {'Authorization' : 'APPCODE %s' % appcode}
        request = urllib2.Request(url = url, headers = headers, data = body)
        try:
            response = urllib2.urlopen(request, timeout = 10)
            return response.code, response.headers, response.read()
        except urllib2.HTTPError as e:
            return e.code, e.headers, e.read()


def demo():
    appcode = '你的APPCODE'
    url = 'http://dm-51.data.aliyun.com/rest/160601/ocr/ocr_idcard.json'
    img_file = '图片文件路径/图片url'
    configure = {'side':'face'}
    #如果没有configure字段，configure设为None
    #configure = None

    if img_file[:4] == 'http':
        img_base64data = img_file
    else:
        img_base64data = get_img_base64(img_file)
    stat, header, content = predict( url, appcode, img_base64data, configure)
    if stat != 200:
        print 'Http status code: ', stat
        print 'Error msg in header: ', header['x-ca-error-message'] if 'x-ca-error-message' in header else ''
        print 'Error msg in body: ', content
        exit()
    result_str = content

    print result_str;
    #result = json.loads(result_str)

if __name__ == '__main__':
    demo()

File: c#/appcode_demo.cs
using System;
using System.IO;
using System.Text;
using System.Net;
using System.Net.Security;
using System.Security.Cryptography.X509Certificates;

class Demo{
    static void Main(string[] args)
    {
        String url = "http://dm-51.data.aliyun.com/rest/160601/ocr/ocr_idcard.json";
        String appcode = "你自己的AppCode";
        String img_file = "图片路径";
        
        //如果输入带有inputs, 设置为True，否则设为False
        bool is_old_format = false;

        //如果没有configure字段，config设为''
        //String config = '';
        String config = "{\\\"side\\\":\\\"face\\\"}";
        
        String method = "POST";

        String querys = "";

        FileStream fs = new FileStream(img_file, FileMode.Open);
        BinaryReader br = new BinaryReader(fs);
        byte[] contentBytes = br.ReadBytes(Convert.ToInt32(fs.Length));
        String base64 = System.Convert.ToBase64String(contentBytes);
        String bodys;
        if(is_old_format){
            bodys = "{\"inputs\" :" +
                                "[{\"image\" :" + 
                                    "{\"dataType\" : 50," + 
                                     "\"dataValue\" :\"" + base64 + "\"" +
                                     "}";
            if(config.Length > 0){
                bodys += ",\"configure\" :" + 
                                "{\"dataType\" : 50," +
                                 "\"dataValue\" : \"" + config + "\"}" +
                                 "}";
            } 
            bodys +=  "]}";
        }else{
            bodys = "{\"image\":\"" + base64 + "\"";
            if(config.Length > 0){
                bodys += ",\"configure\" :\"" + config + "\"";
            } 
            bodys += "}";
        }
        HttpWebRequest httpRequest = null;
        HttpWebResponse httpResponse = null;

        if (0 < querys.Length)
        {
            url = url + "?" + querys;
        }

        if (url.Contains("https://"))
        {
            ServicePointManager.ServerCertificateValidationCallback = new RemoteCertificateValidationCallback(CheckValidationResult);
            httpRequest = (HttpWebRequest)WebRequest.CreateDefault(new Uri(url));
        }
        else
        {
            httpRequest = (HttpWebRequest)WebRequest.Create(url);
        }
        httpRequest.Method = method;
        httpRequest.Headers.Add("Authorization", "APPCODE " + appcode);
        //根据API的要求，定义相对应的Content-Type
        httpRequest.ContentType = "application/json; charset=UTF-8";
        if (0 < bodys.Length)
        {
            byte[] data = Encoding.UTF8.GetBytes(bodys);
            using (Stream stream = httpRequest.GetRequestStream())
            {
                stream.Write(data, 0, data.Length);
            }
        }
        try
        {
            httpResponse = (HttpWebResponse)httpRequest.GetResponse();
        }
        catch (WebException ex)
        {
            httpResponse = (HttpWebResponse)ex.Response;
        }

        Console.WriteLine(httpResponse.StatusCode);
        Console.WriteLine(httpResponse.Method);
        Console.WriteLine(httpResponse.Headers);
        Stream st = httpResponse.GetResponseStream();
        StreamReader reader = new StreamReader(st, Encoding.GetEncoding("utf-8"));
        Console.WriteLine(reader.ReadToEnd());
        Console.WriteLine("\n");

    }

    public static bool CheckValidationResult(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors errors)
    {
        return true;
    }
}